arctic/store/metadata_store.py

```diff
@@ -0,0 +1,217 @@
+            Example:
+                {'example': ([{}], [datetime.utcnow()])
+        """
+        if not isinstance(collection, dict):
+            raise TypeError('collection must be a dictionary.')
+        requests = []
+        for symbol, (entries, times) in collection.items():
+            if len(entries) != len(times):
+                raise ValueError('Number of entries and number of time stamps do not match.')
+            if self.has_symbol(symbol):
+                self.delete(symbol)
+            # TODO: Add checks to ensure metadata do change across any timestamp
+            for metadata, start_time, end_time in zip(entries, times, times[1:] + [None]):
+                doc = {'symbol': symbol, 'metadata': metadata, 'start_time': start_time}
+                if end_time is not None:
+                    doc['end_time'] = end_time
+                requests.append(doc)
+
+        self.insert_many(requests)
+
+    def write_metadata(self, symbol, metadata, start_time=None):
+        """
+        Update metadata entry for `symbol`
+
+        Parameters
+        ----------
+        symbol : `str`
+            symbol name for the item
+        metadata : `dict`
+            to be persisted
+        start_time : `datetime.datetime`
+            when metadata becomes effective
+            Default: datetime.utcnow()
+        """
+        if start_time is None:
+            start_time = dt.utcnow()
+        old_metadata = self.find_one({'symbol': symbol}, sort=[('start_time', pymongo.DESCENDING)])
+        if old_metadata is not None:
+            if old_metadata['start_time'] >= start_time:
+                raise ValueError('start_time is earlier than the last metadata')
+            if old_metadata['metadata'] == metadata:
+                logger.warning('No change to metadata')
+                return metadata
+        elif metadata is None:
+            return
+
+        self.find_one_and_update({'symbol': symbol}, {'$set': {'end_time': start_time}},
+                                  sort=[('start_time', pymongo.DESCENDING)])
+        return self._write_metadata_entry(symbol, metadata, start_time)
+
+    def delete_last_metadata(self, symbol):
+        """
+        Delete current metadata of `symbol`
+
+        Parameters
+        ----------
+        symbol : `str`
+            symbol name to delete
+
+        Returns
+        -------
+        Deleted metadata
+        """
+        last_metadata = self.find_one({'symbol': symbol}, sort=[('start_time', pymongo.DESCENDING)])
+        if last_metadata is None:
+            raise NoDataFoundException('No metadata found')
+
+        self.find_one_and_delete({'symbol': symbol}, sort=[('start_time', pymongo.DESCENDING)])
+        mongo_retry(self.find_one_and_update)({'symbol': symbol}, {'$unset': {'end_time': ''}},
+                                              sort=[('start_time', pymongo.DESCENDING)])
+
+        return last_metadata
+
+    def delete_metadata(self, symbol):
```

Comment:
May be `delete_all_metadata`? `delete_metadata` sounds quite innocent for what it actually does (chop everything)